src/main/kotlin/com/doist/gradle/changelog/ChangelogProcessor.kt

```diff
@@ -0,0 +1,34 @@
+package com.doist.gradle.changelog
+
+import java.io.File
+
+internal class ChangelogProcessor(private val config: CommitConfig) {
+
+    fun collectPendingEntries(file: File): List<String> {
+        return file.walk()
+            .filter { it.isFile }
+            .map { it.readLines() }
+            .flatten()
+            .filter { it.isNotBlank() }
+            .toList()
+    }
+
+    fun insertEntries(changelogFile: File, entries: List<String>) {
+        if (entries.isEmpty()) return
+
+        val changelog = changelogFile.readLines().toMutableList()
+        val prefix = config.prefix?.plus("\n") ?: ""
+        val postfix = config.postfix?.plus("\n") ?: ""
+        val entryPrefix = config.entryPrefix ?: ""
+        val entryPostfix = config.entryPostfix ?: ""
```

Comment:
Probably it's better to add `- ` to `entryPrefix` or `\n` to `entryPostfix`, otherwise by default entries will be rendered as a single paragraph without line breaks.

Reply:
Not sure I understand 🤔

For example, if the file is empty, and we insert `listOf("Feature 1", "Feature 2")` with a default config, the result will be:
```
Feature 1
Feature 2
```
So one entry per line.

I'll add a test to make it easier to play with 👍